cup-maven-plugin/src/main/java/de/jflex/plugin/cup/GenerateMojo.java

```diff
@@ -0,0 +1,196 @@
+package de.jflex.plugin.cup;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Optional;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugin.logging.Log;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+
+/** Creates a Java parser from CUP definition, using CUP. */
+@Mojo(name = "generate", defaultPhase = LifecyclePhase.GENERATE_SOURCES, threadSafe = false)
+public class GenerateMojo extends AbstractMojo {
+
+  /** Constant {@code src/main/cup}. */
+  private static final String DEFAULT_SRC_DIRECTORY = "src/main/cup";
+
+  /** In a CUP definition, the Java package is introduce by the {@code package} keyword. */
+  private static final String PACKAGE_DEFINITION = "package";
+
+  private static final String DEFAULT_JAVA_PACKAGE = "";
+
+  // Seriously – by default CUP generates lower case class names.
```

Comment:
I understand the exasperation, but I'd still remove the 'Seriously' ;-)

Reply:
Done.